server/src/main/java/org/opensearch/index/codec/composite/composite99/Composite99DocValuesWriter.java

```diff
@@ -243,16 +256,29 @@ public NumericDocValues getNumeric(FieldInfo field) {
                 }
             });
         } else {
-            fieldProducerMap.put(compositeField, new EmptyDocValuesProducer() {
-                @Override
-                public SortedNumericDocValues getSortedNumeric(FieldInfo field) {
-                    return DocValues.emptySortedNumeric();
-                }
-            });
+            if(isSortedSetField(compositeField)) {
```

Comment:
should we have a switch? for when we support more field types.

Reply:
I don't see anything else for forseeable future apart from sortedSet and sortedNumeric